source/base/subscriptor.cc

```diff
@@ -62,6 +62,22 @@ Subscriptor::Subscriptor (const Subscriptor &)
 {}
 
 
+
+#ifdef DEAL_II_WITH_CXX11
+Subscriptor::Subscriptor (Subscriptor &&subscriptor)
+  :
+  counter(0),
+  object_info (subscriptor.object_info)
+{
+  // Explicitly invoke the destructor of `Subscriptor` for the object
+  // to be moved from in order to guarantee that we're not moving an
+  // object that has subscriptions.
+  subscriptor.~Subscriptor();
```

Comment:
Ok, I'll change it. I could factor the error handling out into a separate function called from both methods in order to avoid duplicated code, if you think that's best.